config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/BundleMetadataBuilder.java

```diff
@@ -135,11 +130,12 @@ private BundleMetadata buildFullBundleMetadata(final List<Entity> entities, fina
 
     private Collection<Metadata> getDefinedEntitiesMetadata(final List<Entity> definedEntities) {
         return definedEntities.stream().filter(FILTER_NON_ENV_ENTITIES_EXCLUDING_FOLDER)
-                .map(e -> ((GatewayEntity)e.getGatewayEntity()).getMetadata()).collect(Collectors.toList());
+                .map(e -> (e.getGatewayEntity()).getMetadata()).collect(Collectors.toList());
     }
 
-    private boolean isBundleContainsReusableEntity (final AnnotatedBundle annotatedBundle) {
-        return annotatedBundle.getAnnotatedEntity().isReusable();
+    private boolean isBundleContainsSharedEntity(final AnnotatedBundle annotatedBundle) {
```

Comment:
why do we have to check individual policy or encass, if annotation is there then it is shared right